### PR TITLE
Set up plot testing infrastructure and tests for plot_coverage

### PR DESCRIPTION
**Pull request by sarah-colq**
a) set up `DEVELOP.md` and subdirectories for using `vdiffr` to test plots.
b) Add tests for `plot_coverage` (not `one` type, see issue #516)
c) move `compute_coverage` and `compute_calibration` to their plot files because that's the only place those functions are used (to emulate how `plot_width` is set up).
d) move `test_legal_faceting` and `Interaction(...)` to `grp_processing_for_facets.R` since those are used by multiple plots.  Also add a test for `test_legal_faceting`.
e) rename `test-evaluate.R` to `test-evaluate_predictions.R` because the main file was renamed as well.

### CODE REVIEW

File: R-packages/evalcast/DEVELOP.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+relative to the `tests/testthat/` directory regardless of your current working
+directory, so for example you can use
+
+```r
+foo <- readRDS(test_path("data/foo.rds"))
+```
+
+to load `tests/testthat/data/foo.rds`.
+
+## Documentation
```
Comment (dajmcdon) [inline R-packages/evalcast/DEVELOP.md:72]: This is great!!!

File: R-packages/evalcast/tests/testthat/test-grp_processing.R [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  expect_equal(test_legal_faceting(NULL, "bar", c("foo", "bar")), TRUE)
+  expect_equal(test_legal_faceting("foo", "bar", c("foo", "bar")), TRUE)
+  
+  expect_error(
+    test_legal_faceting("foo", "bar", "bar"),
+    "Variables must be grouped in order to be faceted in rows: foo")
+  expect_error(
+    test_legal_faceting("foo", "bar", "foo"),
+    "Variables must be grouped in order to be faceted in cols: bar")
+})
```
Comment (dajmcdon) [inline R-packages/evalcast/tests/testthat/test-grp_processing.R:14]: Need a trailing new line here.

File: R-packages/evalcast/tests/testthat/test-plot_coverage.R [hunk 3]
```diff
@@ -0,0 +1,169 @@
+library(testthat)
```
Comment (dajmcdon) [inline R-packages/evalcast/tests/testthat/test-plot_coverage.R:1]: shouldn't need `library(testthat)` I don't think.

**Review by dajmcdon — CHANGES_REQUESTED** [review]

File: R-packages/evalcast/DEVELOP.md (hunk 1 above)
Comment (dajmcdon) [reply R-packages/evalcast/DEVELOP.md:72]: Comment applies to the entire file!

File: R-packages/evalcast/DESCRIPTION [hunk 4]
```diff
@@ -84,6 +84,7 @@ Suggests:
     testthat,
     mockr,
     mockery,
-    scoringutils
+    scoringutils,
+    vdiffr
```
[the remark this answers is not in the document]
Comment (sarah-colq) [reply R-packages/evalcast/DESCRIPTION:88]: Already in the suggests!

File: R-packages/evalcast/DEVELOP.md (hunk 1 above)
Comment (sarah-colq) [reply R-packages/evalcast/DEVELOP.md:72]: Thanks :D

**Pull request merged**